src/Components/OurWork/OurWork.tsx

```diff
@@ -1,7 +1,119 @@
+import React, { useState } from 'react';
+import workData from './data';
+import { Tabs } from './Tabs';
+import { Button } from 'antd';
+import { RightOutlined } from '@ant-design/icons';
+import { Category } from './Category';
+import { ShowMore } from './ShowMore';
 import './style.scss';
 
-const OurWork = () => {
-  return <div></div>;
+const OurWork: React.FC = () => {
+  const [activeTab, setActiveTab] = useState<'Designs' | 'Case studies'>(
+    'Designs'
+  );
+  const [activeDivisionButton, setActiveDivisionButton] =
+    useState<string>('All');
+  const [visibleItemsCount, setVisibleItemsCount] = useState<{
+    [key: string]: number;
+  }>({
+    'Designs-All': 4,
+    'Designs-UI/UX design': 4,
+    'Designs-Branding': 4,
+    'Designs-Graphic design': 4,
+    'Designs-3D modelling': 4,
+    'Designs-Video production': 4,
+    'Case studies-All': 4,
+  });
+
+  const currentKey = `${activeTab}-${activeDivisionButton}`;
+  const handleSetVisibleItemsCount = (count: number) => {
+    setVisibleItemsCount(prevCounts => ({
+      ...prevCounts,
+      [currentKey]: count,
+    }));
+  };
+
+  const handleTabChange = (newTab: 'Designs' | 'Case studies') => {
+    setActiveTab(newTab);
+    setActiveDivisionButton('All');
+  };
+  const filteredData = workData.filter(item => {
+    const matchesTabs =
+      activeTab === 'Designs'
+        ? item.category === 'Designs'
+        : item.category === 'Case studies';
+
+    const matchesCategories =
+      activeDivisionButton.toLowerCase() === 'all' ||
+      (item.division &&
+        item.division.toLowerCase() === activeDivisionButton.toLowerCase());
+
+    return matchesTabs && matchesCategories;
+  });
+
+  const visibleData = filteredData.slice(0, visibleItemsCount[currentKey] || 4);
+  const isShowingAll = visibleItemsCount[currentKey] >= filteredData.length;
```

Comment:
good practice to add type for all constants
```suggestion
  const isShowingAll:boolean = visibleItemsCount[currentKey] >= filteredData.length;
```

Reply:
Added